frontend/packages/console-shared/src/components/formik-fields/EnvironmentField.tsx

```diff
@@ -1,27 +1,75 @@
 import * as React from 'react';
+import * as _ from 'lodash';
 import { useFormikContext, FormikValues } from 'formik';
-import { EnvironmentPage } from '@console/internal/components/environment';
+import { NameValueEditor } from '@console/internal/components/utils/name-value-editor';
 import { FormGroup } from '@patternfly/react-core';
-import { EnvironmentFieldProps, NameValuePair, NameValueFromPair } from './field-types';
+import { EnvironmentFieldProps } from './field-types';
 import { getFieldId } from './field-utils';
+import { SecretModel, ConfigMapModel } from '@console/internal/models';
+import { k8sGet } from '@console/internal/module/k8s';
 
 const EnvironmentField: React.FC<EnvironmentFieldProps> = ({
   label,
   helpText,
   required,
+  envs,
   ...props
 }) => {
   const { setFieldValue } = useFormikContext<FormikValues>();
   const fieldId = getFieldId(props.name, 'env-input');
+  const [nameValue, setNameValue] = React.useState(!_.isEmpty(envs) ? envs : [['', '']]);
+  const [configMaps, setConfigMaps] = React.useState({});
+  const [secrets, setSecrets] = React.useState({});
+  const nsRef = React.useRef({ namespace: props.obj?.metadata?.namespace });
+  nsRef.current = { namespace: props.obj?.metadata?.namespace };
+
+  const handleNameValuePairs = React.useCallback(
+    ({ nameValuePairs }) => {
+      const updatedNameValuePairs = _.compact(
+        nameValuePairs.map(([name, value]) => {
+          if (_.isObject(value)) {
+            return { name, valueFrom: value };
+          }
+          if (value.length) {
+            return { name, value };
+          }
+          return null;
+        }),
+      );
+      setNameValue(nameValuePairs);
+      setFieldValue(props.name, updatedNameValuePairs);
+    },
+    [props.name, setFieldValue],
+  );
+  React.useEffect(() => {
+    Promise.all([
+      k8sGet(ConfigMapModel, null, nsRef.current.namespace),
+      k8sGet(SecretModel, null, nsRef.current.namespace),
+    ])
+      .then(([nsConfigMaps, nsSecrets]) => {
+        setConfigMaps(nsConfigMaps);
+        setSecrets(nsSecrets);
+      })
+      .catch((err) => {
+        if (err?.response?.status !== 403) {
+          throw new Error(err?.message);
```

Comment:
This simply results in a console log where a user won't know what's going on. Maybe we're better off displaying an error modal as is done in most places.
```
Uncaught (in promise) Error: <message>
    at EnvironmentField.tsx:56
```

Modal:
```
      errorModal({ error: err?.message });
```

Reply:
Using modal now.